4_gap_analysis/src/1c_N_byState.R

```diff
@@ -0,0 +1,205 @@
+             'Total Phosphorus, mixed forms',NA)}
+
+if (Con=='TKN'){
+  ConList<-c('Kjeldahl nitrogen','Kjeldahl nitrogen as N','Nitrogen, Ammonia + Organic','Nitrogen, Kjeldahl',
+             'Total Kjeldahl nitrogen','Total Kjeldahl nitrogen (Organic N & NH3)')}
+
+# Crosswalk based on Harmonization table
+xWalk1<-loadWorkbook('4_gap_analysis/in/NH2012Trends_N.xlsx') %>% readWorksheet(sheet='data_harmonization',header=T) %>%
+  filter(orig_parameter_name %in% ConList,qwdata_source %in% c('STORET/WQX','WQP'))
+
+# Crosswalk based on Agency name, NDA acroym, and WQX agency code
+xWalk2<-loadWorkbook('4_gap_analysis/in/Organizations_WQP.xlsx') %>% readWorksheet(sheet='WQP_Organizations_mlr',header=T)
+
+#******************************
+# Reading files locally
+all_files <- dir('1_wqpdata/out/byState', pattern=paste0("^",state),full.names=T) %>% #pattern=paste0('*',state,'*'),full.names = T) %>%
+  grep(paste('(',Con,')',sep=''), ., value=TRUE)
+# Add ortho P data to the phosphorus files
+if(Con=='phosphorus'){
+  orthoPfiles<-dir('1_wqpdata/out/byState', pattern=paste0(state,'*'),full.names = T) %>%
+    grep('(orthophosphate)', ., value=TRUE)
+  all_files<-append(all_files,orthoPfiles)
+}
+data_files<-all_files[!grepl('siteinfo|ind',all_files)] # just the data files
+site_files<-all_files[grepl('siteinfo|ind',all_files)] # just the site files
+
+# combine into one data frame
+combined_data <- bind_rows(
+  lapply(data_files, function(dfile) {
+    # By State
+    dat <- feather::read_feather(dfile) %>%
+      # Below here are data coercion fixes for columns
+      mutate_at(vars(starts_with('ResultDepthHeightMeasure.MeasureValue'),starts_with('ActivityTopDepthHeightMeasure.MeasureValue'),
+                     starts_with('ActivityBottomDepthHeightMeasure.MeasureValue'),starts_with('ResultAnalyticalMethod.MethodIdentifier'),
+                     starts_with('ProjectIdentifier'),starts_with('SampleCollectionMethod.MethodIdentifier'),
+                     starts_with('ActivityCommentText'),starts_with('LaboratoryName'),starts_with('ActivityDepthAltitudeReferencePointText')),
+                list(as.character)) %>%
+      mutate(ConstituentGroup=Con) %>%
+      #See Read me for explanations on each constituent
+      filter(!grepl('Quality Control',ActivityTypeCode) & ActivityMediaName %in% c('Water','water','Sediment') &
+                ActivityMediaSubdivisionName %in% c('Surface Water',NA,'','Unknown') &
+                !grepl('%',ResultMeasure.MeasureUnitCode) &
+                ActivityStartDate > as.Date(StartDate) & ActivityStartDate < as.Date(EndDate))
+  })
+) %>%
+  # Drop date fields if NA
+  filter(!is.na(ActivityStartDate)) %>%
+  mutate(ResultSampleFractionText=replace(ResultSampleFractionText,ResultSampleFractionText==" ",NA))
+
+
+# subset by rows with data values in one of two fields
+combined_data<-subset(combined_data,(!is.na(combined_data$ResultMeasureValue)) |
```

Comment:
Same comment as before - potentially join with chain of `dplyr` commands above and use `filter` instead of `subset`.